hooks/request/request.go

```diff
@@ -83,62 +54,118 @@ func (h *Hook) AfterInitChain(ctx sdk.Context, req abci.RequestInitChain, res ab
 
 // AfterBeginBlock specify actions need to do after begin block period (app.Hook interface).
 func (h *Hook) AfterBeginBlock(ctx sdk.Context, req abci.RequestBeginBlock, res abci.ResponseBeginBlock) {
-	trans, err := h.dbMap.Begin()
-	if err != nil {
-		panic(err)
-	}
+	trans := h.db.Begin()
 	h.trans = trans
 }
 
 // AfterDeliverTx specify actions need to do after transaction has been processed (app.Hook interface).
 func (h *Hook) AfterDeliverTx(ctx sdk.Context, req abci.RequestDeliverTx, res abci.ResponseDeliverTx) {
+	reports := make(map[types.RequestID][]types.Report)
+	for _, event := range res.Events {
+		events := sdk.StringifyEvents([]abci.Event{event})
+		evMap := common.ParseEvents(events)
+
+		switch event.Type {
+		case types.EventTypeReport:
+			reqID := types.RequestID(common.Atoi(evMap[types.EventTypeReport+"."+types.AttributeKeyID][0]))
+			validator := evMap[types.EventTypeReport+"."+types.AttributeKeyValidator][0]
+			iter := h.oracleKeeper.GetReportIterator(ctx, reqID)
+			defer iter.Close()
+
+			// Check all reports related to given request ID
+			for ; iter.Valid(); iter.Next() {
```

Comment:
I think we can write `GetReport` function in keeper to find a report on request id and validator.